mdoc/mdoc.csproj

```diff
@@ -167,4 +173,10 @@
     </ProjectReference>
   </ItemGroup>
   <ItemGroup />
+  <PropertyGroup Condition=" '$(Configuration)' != 'Release' ">

```

Comment:
So, this post build event was added to copy test dlls and Windows specific dlls (needed to run UWP specific code) to the bin folders. Those dlls are not generated by mdoc or mdoc.Test so we put them to the 'external' folder and copy them to bin right after build is finished. 

From our perspective, it's not good solution to commit those external libs directly to the bin folder since they are not generated during mdoc build process. Also committing bin folder will remove this folder from gitignore and all files inside bin folder will be tracked as added/modified by source control system. This will lead to conflicts when we do pull on the repository. Also if someone will clean bin folder manually then mdoc will not run locally until someone does not copy this dlls one more time manually. That's why we prefer to add post build event instead of commit dlls - to protect from issues in the future.   

We re-wrote the post build event to use .sh script instead of using xcopy utility from Windows. So we expect that this will solve your issue. Could you please verify that the issue disappeared for Mac?

Reply:
No, unfortunately. All of those points you mentioned are indeed difficulties, however the main reason that it's also a problem is because the `bin/Release` directory is what gets packaged into the Nuget package. So if these test assemblies are there, they will be distributed needlessly via nuget. 

Here's what I'd like to see:

- building `apidoctools.sln` will _not_ result in the building of any other test assemblies.
- the windows-only test assemblies will be compiled on windows, and checked into some folder in the `mdoc/Test` folder.
- There should be a separate test-only solution that compiles these test assemblies ... so that if they are to be changed in the future, the developer can open that solution on windows, recompile them after changes are made, and check in the newly-changed binaries.

Reply:
We do not copy test dlls to the mdoc, only to the mdoc.Test. So, the trouble is that we need to copy not  also Windows-related libs. Let me explain in more details.

Currently, we need to copy two types of libs:

1. test libs - currently they are committed to the 'external\Test' folder; those libs are used for unit testing since they cannot be built in usual way; they include c++ and UWP test data.
2. Windows libs - currently they are committed to the 'external\Windows' folder; those libs are used by mdoc and mdoc.Test for processing UWP-specific types. They are not referenced to any project because they are not used explicitly. However, they are loaded dynamically in runtime (I guess, by Mono.Cecil) when UWP-specific type in processed in mdoc. Those libs needs to be located in the 'bin' folder to allow mdoc to retrieve UWP-specific type.

Talking about your proposal, it is already done in the similar manner - c++ and UWP test libs are build manually and committed to the 'external\Test' folder. After that, they are copied to the bin of mdoc.Test to be used during unit testing.
@joelmartinez Do I understand you correctly that I need to relocate them to the 'mdoc/Test' folder, update path to the test dlls in mdoc.Tests and remove copying those dlls from post-build event? 

And talking about Windows libs, they needs to be put to the bin folder of both mdoc and mdoc.Test to be able to process UWP-specific libs. And I assume, that Windows-specific libs need to be put to the Nuget package to prevent mdoc from runtime errors.
@joelmartinez Do you feel like at the end I need to copy Windows libs to the 'bin/Release' folder of mdoc despite of the difficulties I mentioned in previous comment?

Reply:
ahh, I see ... the picture is becoming clearer now. Can point to me the mechanism in mdoc that loads these assemblies at runtime? I briefly skimmed through the code and I can't see where it's loading those.

Reply:
We do not do any calls to those libs directly through the code. They are not even referenced to the mdoc project. The loading happens in runtime automatically and is managed somehow by the system (I guess, it is Mono.Cecil which requires additional info about types). Windows libs are  loaded to the memory only when you process UWP-specific lib and are not required for other.

Reply:
Ahh, so the assembly is loaded by the AssemblyLoader! Ok, so this is perfectly fine for the unit tests. however it is *critical* that these placeholder assemblies are not provided at runtime. The person running `mdoc` must provide the path to the UWP dependencies at runtime (either by passing `-L` on the command line, or in the `dependencies` xml node in `frameworks.xml`). So now that we've gotten clarification on this:

- It's totally fine for the unit test(s) to copy these resources for testing ... however you can easily add a file reference to the assembly in the unit test project, and set it to `Copy if Newer`. That way, msbuild will take care of copying the file without the need for a manually written `.sh` script. Similarly, if these are needed during the integration tests at all, you can just pass the path to the directory that contains the placeholder assemblies via `-L`
- Please make sure that the `bin/Release` folder _does not_ contain the windows/uwp assemblies ... because ideally, anyone documenting UWP libraries will pass the appropriate dependency folder for where the system libs are installed.

I'm sorry for all the confusion and back and forth ... but this in particular is very important to get right. Thanks so much for all the work you've done on this so far!

Reply:
Done.
 So, according to our discussion:
- when UWP library is going to be processed by mdoc, there is a need to specify folder with Windows-libs in a command line (eg, mdoc.exe update -lang c++/winrt **-lib D:\WORK\Microsoft\mdoc\external\Windows**  -o en UwpTestWinRtComponentCpp.winmd)
- for other types of dll Windows libs are not mandatory